plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/snapshot/LinstorVMSnapshotStrategy.java

```diff
@@ -0,0 +1,347 @@
+        if (snapshotMemory) {
+            return StrategyPriority.CANT_HANDLE;
+        }
+        return allVolumesOnLinstor(vmId);
+    }
+
+    @Override
+    public StrategyPriority canHandle(VMSnapshot vmSnapshot) {
+        VMSnapshotVO vmSnapshotVO = (VMSnapshotVO) vmSnapshot;
+        if (vmSnapshotVO.getType() != VMSnapshot.Type.Disk) {
+            return StrategyPriority.CANT_HANDLE;
+        }
+        return allVolumesOnLinstor(vmSnapshot.getVmId());
+    }
+
+    private StrategyPriority allVolumesOnLinstor(Long vmId) {
+        List<VolumeObjectTO> volumeTOs = _vmSnapshotHelper.getVolumeTOList(vmId);
+        if (volumeTOs == null || volumeTOs.isEmpty()) {
+            return StrategyPriority.CANT_HANDLE;
+        }
+        for (VolumeObjectTO volumeTO : volumeTOs) {
+            Long poolId = volumeTO.getPoolId();
+            StoragePoolVO pool = _storagePoolDao.findById(poolId);
+            if (!pool.getStorageProviderName().equals(LinstorUtil.PROVIDER_NAME)) {
+                return StrategyPriority.CANT_HANDLE;
+            }
+        }
+        return StrategyPriority.HIGHEST;
+    }
+
+    private String linstorDeleteSnapshot(final DevelopersApi api, final String rscName, final String snapshotName) {
+        String resultMsg = null;
+        try {
+            ApiCallRcList answers = api.resourceSnapshotDelete(rscName, snapshotName, Collections.emptyList());
+            if (answers.hasError()) {
+                resultMsg = LinstorUtil.getBestErrorMessage(answers);
+            }
+        } catch (ApiException apiEx) {
+            log.error("Linstor: ApiEx - " + apiEx.getBestMessage());
+            resultMsg = apiEx.getBestMessage();
+        }
+
+        return resultMsg;
+    }
+
+    @Override
+    public boolean deleteVMSnapshot(VMSnapshot vmSnapshot) {
+        UserVmVO userVm = _userVmDao.findById(vmSnapshot.getVmId());
+        VMSnapshotVO vmSnapshotVO = (VMSnapshotVO) vmSnapshot;
+        try {
+            _vmSnapshotHelper.vmSnapshotStateTransitTo(vmSnapshot, VMSnapshot.Event.ExpungeRequested);
+        } catch (NoTransitionException e) {
+            log.debug("Failed to change vm snapshot state with event ExpungeRequested");
+            throw new CloudRuntimeException(
+                    "Failed to change vm snapshot state with event ExpungeRequested: " + e.getMessage());
+        }
+
+        List<VolumeObjectTO> volumeTOs = _vmSnapshotHelper.getVolumeTOList(vmSnapshot.getVmId());
+        final StoragePoolVO storagePool = _storagePoolDao.findById(volumeTOs.get(0).getPoolId());
+        final DevelopersApi api = LinstorUtil.getLinstorAPI(storagePool.getHostAddress());
+
+        final String snapshotName = vmSnapshotVO.getName();
+        for (VolumeObjectTO volumeObjectTO : volumeTOs) {
+            final String rscName = LinstorUtil.RSC_PREFIX + volumeObjectTO.getUuid();
+            String err = linstorDeleteSnapshot(api, rscName, snapshotName);
```

Comment:
Hi @rp-, CloudStack allows you to create VM snapshots with the same name. When you have multiple VM snapshots on the same machine, could this cause some unexpected behaviour when you create/delete/revert VM snapshots?

Reply:
Are you referring to the display name?
I don't use that as the snapshot name within Linstor, but rather the `"Name"` e.g.: `i-2-11-VM_VS_20240321102943`
I hope/guess they are kinda unique?

Reply:
yes, sorry, I just saw that the name is set here with a timestamp and it doesn't refer to the display name

https://github.com/apache/cloudstack/blob/502a633ddd0101ebac4460dad6457bc74b2b28ee/server/src/main/java/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java#L349

Reply:
No worries, thanks for looking.